codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketCloseStatus.java

```diff
@@ -0,0 +1,238 @@
+    public static final WebSocketCloseStatus INVALID_PAYLOAD_DATA = register(1007, "Invalid payload data");
+    public static final WebSocketCloseStatus POLICY_VIOLATION = register(1008, "Policy violation");
+    public static final WebSocketCloseStatus MESSAGE_TOO_BIG = register(1009, "Message too big");
+    public static final WebSocketCloseStatus MANDATORY_EXTENSION = register(1010, "Mandatory extension");
+    public static final WebSocketCloseStatus INTERNAL_SERVER_ERROR = register(1011, "Internal server error");
+
+    // 1004, 1005, 1006, 1015 are reserved and should never be used by user
+    //public static final WebSocketCloseStatus SPECIFIC_MEANING = register(1004, "...");
+    //public static final WebSocketCloseStatus EMPTY = register(1005, "Empty");
+    //public static final WebSocketCloseStatus ABNORMAL_CLOSURE = register(1006, "Abnormal closure");
+    //public static final WebSocketCloseStatus TLS_HANDSHAKE_FAILED(1015, "TLS handshake failed");
+
+    private final int statusCode;
+    private final String reasonText;
+
+    private WebSocketCloseStatus(int statusCode, String reasonText) {
+        if (!isValidRfc6455Code(statusCode)) {
+            throw new IllegalArgumentException(
+                "WebSocket close status code does NOT comply with RFC-6455: " + statusCode);
+        }
+        if (reasonText == null) {
+            throw new NullPointerException("Reason text cannot be null");
+        }
+        this.statusCode = statusCode;
+        this.reasonText = reasonText;
+    }
+
+    public int code() {
+        return statusCode;
+    }
+
+    public String reasonText() {
+        return reasonText;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (null == o || getClass() != o.getClass()) {
+            return false;
+        }
+
+        WebSocketCloseStatus that = (WebSocketCloseStatus) o;
+
+        return statusCode == that.statusCode;
+    }
+
+    @Override
+    public int hashCode() {
+        return statusCode;
+    }
+
+    @Override
+    public String toString() {
+        // E.g.: "1000 Bye", "1009 Message too big"
+        return code() + " " + reasonText();
+    }
+
+    public static boolean isValidRfc6455Code(int code) {
+        return code < 0 ||
+            1000 <= code && code <= 1003 ||
+            1007 <= code && code <= 1011 ||
+            3000 <= code;
+    }
+
+    public static WebSocketCloseStatus valueOf(int code) {
+        WebSocketCloseStatus status = VALUES.get(code);
+        if (status == null) {
+            status = new WebSocketCloseStatus(code, "Close status #" + code);
+            VALUES.putIfAbsent(code, status);
+        }
+        return status;
+    }
+
+    public static WebSocketCloseStatus register(int code, String reasonText) {
+        WebSocketCloseStatus status = new WebSocketCloseStatus(code, reasonText);
+        VALUES.put(code, status);
```

Comment:
I am not sure how I feel about overriding stuff in here ... @trustin @carl-mastrangelo @ejona86 thoughts ?

Reply:
fixed as suggested by @trustin